File: do_not_commit/eval-detected.js
/**
 * Only report `eval` when we provide it with non-constant parameters.
 */

/**
 * Negative matches
 */

// ok
eval('var x = "static strings are okay";');

// ok
const constVar = "function staticStrings() { return 'static strings are okay';}";
eval(constVar);

// ok - const within another const
eval(`${constVar}`);

// ok - concatenating with another const okay
const secondConstVar = 'this is a const variable';
eval(constVar + secondConstVar);

/**
 * Positive Matches
 */

let dynamic = "function dynamicStrings() { return 'dynamic strings are not'; }"

// ruleid:eval-detected
eval(dynamic + 'possibly malicious code');

// ruleid:eval-detected
eval(`${dynamic} possibly malicious code`);

// ruleid:eval-detected
eval(dynamic.concat(''));

function evalSomething(something) {
    // ruleid:eval-detected
    eval(something);
}

File: do_not_commit/eval-detected.py
# ok
eval("x = 1; x = x + 2")

blah = "import requests; r = requests.get('https://example.com')"
# ok
eval(blah)

dynamic = "import requests; r = requests.get('{}')"
# ruleid:eval-detected
eval(dynamic.format("https://example.com"))

def eval_something(something):
    # ruleid:eval-detected
    eval(something)

from something import eval

# ok
eval("something")
